src/pages/index.js

```diff
@@ -90,44 +98,92 @@ const profileEditPopup = new PopupWithForm(
 profileEditPopup.setEventListeners();
 
 // Profile Edit Popup Button
+const profileEditButton = document.querySelector("#profile-edit-button");
+
 profileEditButton.addEventListener("click", () => {
-  const userInfoData = userInfo.getUserInfo();
+  if (userInfoInstance) {
+    const userInfoData = userInfoInstance.getUserInfo();
 
-  profileTitleInput.value = userInfoData.name;
-  profileDescriptionInput.value = userInfoData.job;
+    profileTitleInput.value = userInfoData.name;
+    profileDescriptionInput.value = userInfoData.job;
 
-  profileEditPopup.open();
+    profileFormValidator.resetValidation();
+    profileEditPopup.open();
+  }
 });
 
-// Profile Changes Save
 function handleProfileSaveSubmit(inputValues) {
-  userInfo.setUserInfo({
+  const saveButton = document.getElementById("profile-modal-save-button");
+
+  const newInfo = {
     name: inputValues.title,
-    job: inputValues.description,
-  });
+    about: inputValues.description,
+  };
+
+  saveButton.textContent = "Saving...";
+
+  api
+    .editUserInfo(newInfo)
+    .then((res) => {
+      console.log(res);
+      userInfoInstance.setUserInfo({
+        name: res.name,
+        job: res.about,
+        avatar: res.avatar,
+      });
+    })
+    .catch((error) => {
+      console.error(error);
+    })
+    .finally(() => {
+      saveButton.textContent = "Save";
+    });
 }
 
-// ************************** Preview Image ************************** //
+// *********************** Profile Avatar Edit *********************** //
 
-// Preview Image Popup
-const previewImagePopup = new PopupWithImage({
-  popupSelector: "#preview-image-modal",
-});
+const avatarModal = new PopupWithForm(
+  {
+    popupSelector: "#avatar-edit-modal",
+  },
+  handleAvatarFormSubmit
+);
 
-// Clicking an Image to get the preview Image
-function handleImageClick(cardData) {
-  previewImagePopup.open(cardData);
-}
+avatarModal.setEventListeners();
 
-previewImagePopup.setEventListeners();
+const avatarEditButton = document.getElementById("avatar-image");
 
-// Render Card Function
-const renderCard = (card) => {
-  const cardInstance = new Card(card, "#card-template", handleImageClick);
+avatarEditButton.addEventListener("click", () => {
+  editAvatarFormValidator.resetValidation();
 
-  const cardElement = cardInstance.generateCard();
-  cardSection.addItem(cardElement);
-};
+  avatarModal.open();
+});
+
+function handleAvatarFormSubmit(formData) {
+  const saveButton = document.getElementById("avatar-modal-save-button");
+  const avatarLink = formData.url;
+  avatarImg.src = avatarLink;
```

Comment:
Please, delete it. Because you set everything in line 175

You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change